src/main/java/com/github/webicitybrowser/webicity/renderer/backend/html/HTMLRendererBackendParserSettings.java

```diff
@@ -36,51 +38,16 @@ public CharacterReferenceLookup getUnicodeLookup() {
 
 	@Override
 	public void onNodePopped(Node node) {
-		if (
+		if(
 			node instanceof Element element &&
-			element.getLocalName().equals("style") &&
 			element.getNamespace().equals(Namespace.HTML_NAMESPACE)
 		) {
-			addStylesheet(element, createRulesFromStyleTag(element));
-		} else if( node instanceof Element element &&
-				element.getLocalName().equals("link") &&
-				element.getNamespace().equals(Namespace.HTML_NAMESPACE)
-		) {
-			addStylesheet(element, createRulesFromExternalResources(element));
-		}
-	}
-
-	private void addStylesheet(Element element, CSSRuleList ruleList) {
-		CSSStyleSheet styleSheet = () -> ruleList;
-		((HTMLDocument) element.getOwnerDocument()).getStyleSheets().add(styleSheet);
-	}
-
-	private CSSRuleList createRulesFromStyleTag(Element element) {
-		return createRuleList(new StringReader(DOMTextUtil.getChildTextContent(element)));
-	}
-
-	private CSSRuleList createRulesFromExternalResources(Element element) {
-		if(!element.hasAttribute("href") && !element.hasAttribute("imagesrcset")) {
-			throw new RuntimeException("neither href nor imagesrcset is present in link tag");
-		} else if(element.getAttribute("href").isEmpty()) {
-			throw new RuntimeException("href is present but empty");
-		}
-
-		//TODO: add complete error checking
-
-		String href = element.getAttribute("href");
-		return createRuleList(new InputStreamReader(
-			ClassLoader.getSystemClassLoader().getResourceAsStream("stylesheets/" + href)
-		));
-	}
-
-	private CSSRuleList createRuleList(Reader childContextReader) {
-		try {
-			Token[] tokens = CSSTokenizer.create().tokenize(childContextReader);
-			CSSRule[] rules = CSSParser.create().parseAListOfRules(tokens);
-			return CSSRuleList.create(rules);
-		} catch (IOException e) {
-			throw new RuntimeException(e);
+			TagAction tagAction = TagAction.getAction(element.getLocalName());

```

Comment:
See my comments on TagAction.java
- TagAction will be used for more than just stylesheets
- Pass a class called TagActions, instead of doing a static method call here